### ISSUE TITLE
node process hangs on unrelated error when using custom importers

### ISSUE THREAD

**Issue by mattfysh** [sass/node-sass#1048]
The error when thrown doesn't cause the node process to exit
SSCCE:
```javascript
var opts = {data: '@import "x"', importer: function() {}};
require('node-sass').render(opts, function() {});
throw new Error();
```
We were able to work around this using renderSync but our builds are a little slower. Any ideas what could be causing this, or if there's a better workaround available? thanks.

**Comment by mattfysh** [comment]
Found a way to continue using render (and keep build performance) using an uncaughtException listener:
```javascript
var opts = {data: '@import "x"', importer: function() {}};
require('node-sass').render(opts, function() {});
process.on('uncaughtException', function(e) {
	console.error(e.stack);
	process.kill(process.pid, 'SIGINT');
});
throw new Error('test msg');
```
Not the cleanest solution... are there any plans to fix this behavior for node-sass? I noticed #857 about the process hanging with the UV_THREADPOOL_SIZE the only solution given for now.. thanks

**Comment by xzyfer** [comment]
Thanks for this @mattfysh. This is not related to #857. We should catch this error and return a useful error message.

**Comment by saper** [comment]
Yes, we schedule an event and wait on the lock the event handler returns.
The problems is it never returns, because it never gets called - the event loop gets stopped before it happens.

**Comment by saper** [comment]
So, during the next rewrite of the binding code this will be taken into account. For now there is no easy solution (except for `process.kill`). Sorry for that!
